app/src/org/commcare/dalvik/activities/ReportProblemActivity.java

```diff
@@ -40,21 +40,48 @@ public void onClick(View v) {
         sendReportEmail(reportEntry);
         finish();
     }
-    
-    public String buildMessage(String userInput){
-        
-        SharedPreferences prefs = CommCareApplication._().getCurrentApp().getAppPreferences();
-        
-        String username;
+
+    public static String getDomain(){
+        try {
+            SharedPreferences prefs = CommCareApplication._().getCurrentApp().getAppPreferences();
+            return prefs.getString(HttpRequestGenerator.USER_DOMAIN_SUFFIX, "not found");
+        } catch(NullPointerException e){
+            return "Domain not set.";
+        }
+    }
+
+    public static String getPostURL(){
+        try{
+            SharedPreferences prefs = CommCareApplication._().getCurrentApp().getAppPreferences();
+            return prefs.getString(HttpRequestGenerator.USER_DOMAIN_SUFFIX, "not found");
+        } catch(NullPointerException e){
+            return "PostURL not set.";
+        }
+    }
+
+    public static String getUser(){
+        try{
+            return CommCareApplication._().getSession().getLoggedInUser().getUsername();
+        } catch(SessionUnavailableException e){
+            return "User not logged in.";
+        }
+    }
+
+    public static String getVersion(){
         try {
-            username = CommCareApplication._().getSession().getLoggedInUser().getUsername();
-        } catch (SessionUnavailableException e) {
-            username = "unknown_username";
+            return CommCareApplication._().getCurrentVersionString();
+        } catch(NullPointerException e){
+            return "Version not set.";
```

Comment:
Yeah actually I think you're right on that one

However the reporting was failing when I tested crashes at certain points during/before install, so I threw this error handling (though  can't remember the exact cause now). I'm actually now leaning towards making these all catch just a generic Exception so that we never fail to report due to crashes in generating the per-user info